picard/acoustid/manager.py

```diff
@@ -82,16 +87,42 @@ def submit(self):
         if not submissions:
             self._check_unsubmitted()
             return
-        log.debug("AcoustID: submitting ...")
+        log.debug("AcoustID: submitting total of %d fingerprints...", len(submissions))
+        self._batch_submit(submissions)
+
+    def _batch_submit(self, submissions, retries=2, unsuccessful=None):
+        if not submissions:  # All fingerprints submitted
+            if unsuccessful:
+                if retries > 0:  # Retry unsuccessful submissions
+                    log.info("AcoustID: failed submitting %d fingerprints, retrying %d times",
+                        len(unsuccessful), retries)
+                    self._batch_submit(unsuccessful, retries - 1)
```

Comment:
What about using a small delay between retries? I think we could use something like:

```python
DELAY_MSEC = 1000
QtCore.QTimer.singleShot(DELAY_MSEC, partial(self._batch_submit, unsuccessful, retries - 1))
```
```

Reply:
There is already a natural delay between submissions. Otherwise I don't think adding a small delay would help much. If this would be a server application running in the background I would say we try again after a few minutes, but this does not really help.

I get the feeling we are trying to solve an issue here we don't really have. The webservice module already does do retries on some kind of errors. Picard is doing a lot of web requests. Every one could fail, we could add retries for all of them. But I don't think it is good client behavior to just rerun requests for no good reason if the server responded with a failure. I'm also not sure what makes AcoustID special that we should try extra hard to get this submitted. If this fails it is either a temporary network issue somewhere in between, but this might also take some time to get resolved, or a more permanent server side issue, which likely won't be solved a second later nor even 10 minutes later.

Thinking of it I'm actually tempted to revert the retry logic here.

Reply:
Ok, thanks for explanations, if reverting retry logic makes things simpler (and retry isn't really needed), I'm for it.